### PR TITLE
Update calling command page

### PR DESCRIPTION
**Pull request by MalloD12**

### CODE REVIEW

File: docs/extensions-integrations/integration-guides/calling-commands.md [hunk 1]
```diff
@@ -7,11 +7,81 @@ The API provides both a way to call individual commands and metadata about each
...
+This just a constant where we will define the known names for a given command. 
+
+```java
+public static String[] COMMAND_NAME = {"update"};
+```
+
+Similarly, for certain commands there could be a legacy command name which can be defined in the same way and be passed to the `CommandBuilder` constructor.
+
+### - Arguments:
+There required and optional arguments used by a command that are defined in the below way. Either in the code or [documentation](https://docs.liquibase.com) you can see
```
Comment (filipelautert) [inline docs/extensions-integrations/integration-guides/calling-commands.md:25]: There `are `

File: docs/extensions-integrations/integration-guides/calling-commands.md [hunk 2]
```diff
@@ -7,11 +7,81 @@ The API provides both a way to call individual commands and metadata about each
...
 
-## API Documentation
+# How to use Command Framework
 
-For a complete description of the CommandScope API, including methods to call, see [liquibase.command.CommandScope](../../code/api/command-commandscope.md).
+Main components from Command frameworks you need to know before using it are:
+
+### - Command Name:
+
+This just a constant where we will define the known names for a given command.
```
Comment (filipelautert) [inline docs/extensions-integrations/integration-guides/calling-commands.md:16]: This `is`

File: docs/extensions-integrations/integration-guides/calling-commands.md [hunk 3]
```diff
@@ -7,11 +7,81 @@ The API provides both a way to call individual commands and metadata about each
...
+Dependencies are not useful for calling a command, but they are useful when defining a new command. These specified dependencies are basically part of the different stages of the pipeline execution for a given command.
+```java
+@Override
+    public List<Class<?>> requiredDependencies() {
+        List<Class<?>> deps = Arrays.asList(Database.class, DatabaseChangeLog.class, 
+        ChangeExecListener.class, ChangeLogParameters.class, UpdateSummaryEnum.class);
+        return deps;
+    }
+```
+What we need to understand here is the above dependencies classes will execute their own `run()` method which is needed to accomplish with the `run()` of the given command, for example example update command.
```
Comment (filipelautert) [inline docs/extensions-integrations/integration-guides/calling-commands.md:64]: double example

File: docs/extensions-integrations/integration-guides/calling-commands.md [hunk 4]
```diff
@@ -25,13 +95,71 @@ public class ExampleIntegration  {
...
+            setDatabaseArgumentsToCommand(updateCommand);
+            updateCommand.execute();
         } catch (CommandExecutionException e) {
             System.out.println("Error running update: "+e.getMessage());
         }
+    }
+}
+```
+
+### Execution in scope example
```
Comment (filipelautert) [inline docs/extensions-integrations/integration-guides/calling-commands.md:113]: Why would I execute a command in a scope? What can I do with a scope?

File: docs/extensions-integrations/integration-guides/calling-commands.md [hunk 5]
```diff
@@ -25,13 +95,71 @@ public class ExampleIntegration  {
...
+import liquibase.command.CommandScope;
+import liquibase.exception.CommandExecutionException;
+
+public class ExampleIntegration  {
 
+    public static void main(String[] args) throws Exception {
+        // ... setup configuration system and file access
+
+        try {
+            runInScope(() -> {
```
Comment (filipelautert) [inline docs/extensions-integrations/integration-guides/calling-commands.md:127]: the runInScope method is from liquibase class.. I think you need to use the scope.child here....

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: docs/extensions-integrations/integration-guides/calling-commands.md (hunk 1 above)
Comment (MalloD12) [reply docs/extensions-integrations/integration-guides/calling-commands.md:25]: updated

File: docs/extensions-integrations/integration-guides/calling-commands.md (hunk 2 above)
Comment (MalloD12) [reply docs/extensions-integrations/integration-guides/calling-commands.md:16]: updated

File: docs/extensions-integrations/integration-guides/calling-commands.md (hunk 3 above)
Comment (MalloD12) [reply docs/extensions-integrations/integration-guides/calling-commands.md:64]: updated

File: docs/extensions-integrations/integration-guides/calling-commands.md (hunk 4 above)
Comment (MalloD12) [reply docs/extensions-integrations/integration-guides/calling-commands.md:113]: I thought this way and `Scope.child()` are at the end a similar thing so we can run any of these ways and set different scope settings. For example, a resource accessor, an env. variable, etc.

Comment (filipelautert) [reply docs/extensions-integrations/integration-guides/calling-commands.md:113]: Sorry, I mean... we are using the Scope here but there is no explanation why we are doing that . I think we would need a brief description of what is a Scope and how to use it before starting adding examples using that.

**Review by filipelautert — CHANGES_REQUESTED** [review]

**Review by filipelautert — APPROVED** [review]

**Comment by filipelautert** [comment]
@MalloD12 There is a build failure, could you review it?

**Comment by MalloD12** [comment]
@filipelautert I think now is ok.

**Comment by kevin-atx** [comment]
@MalloD12 and @filipelautert - do you both believe that this update to the contribute docs is ready for release, or are you still reviewing it?

**Comment by MalloD12** [comment]
@kevin-atx in my opinion it would be nice to release so then we start getting feedback from the community and that way improve it as needed.

**Comment by filipelautert** [comment]
@kevin-atx I agree with @MalloD12 !

**Review by kevin-atx — APPROVED** [review]

**Pull request merged**